pyvrp/GeneticAlgorithm.py

```diff
@@ -121,23 +125,31 @@ def run(self, stop: StoppingCriterion):
         return Result(self._best, stats, iters, end)
 
     def _educate(self, individual: Individual):
-        self._ls.search(individual)
+        def _is_new_best(indiv):
+            return indiv.is_feasible() and indiv.cost() < self._best.cost()
 
-        # Only intensify feasible, new best solutions. See also the repair
-        # step below.
-        if (
-            self._params.should_intensify
-            and individual.is_feasible()
-            and individual.cost() < self._best.cost()
-        ):
-            self._ls.intensify(individual)
+        def _add_and_register(indiv):
+            self._pop.add(indiv)
+            self._pm.register_load_feasible(not indiv.has_excess_capacity())
+            self._pm.register_time_feasible(not indiv.has_time_warp())
+
+        intensification_prob = self._params.intensification_probability
+        intensify = self._rng.rand() < intensification_prob
+
+        self._ls.run(individual, intensify)
 
-        if individual.is_feasible() and individual.cost() < self._best.cost():
-            self._best = individual
+        if _is_new_best(individual):
+            self._best = Individual(individual)
 
```

Comment:
GA collecting best individuals is possibly still bugged on main. Since we are not copying the individual, any subsequent intensification or repair step can possibly change the feasibility of the best individual.